### ISSUE TITLE
BLEIO_gatt_read_char_by_uuid Line:155 g_tree_lookup() failed on Raspbian.

### ISSUE THREAD

**Issue by ahidaka** [Azure/azure-iot-gateway-sdk#90]
I checked and completed #51 bluez-5.37 workaround for BLE.

Error: Time:Tue Jan 10 05:45:11 2017 File:/home/pi/q/azure-iot-gateway-sdk/modules/ble/src/ble_gatt_io_linux_read.c Func:BLEIO_gatt_read_char_by_uuid Line:155 g_tree_lookup() failed.
Error: Time:Tue Jan 10 05:45:11 2017 File:/home/pi/q/azure-iot-gateway-sdk/modules/ble/src/bleio_seq_linux_schedule_read.c Func:schedule_read Line:119 BLEIO_gatt_read_char_by_uuid failed with 153.
Error: Time:Tue Jan 10 05:45:11 2017 File:/home/pi/q/azure-iot-gateway-sdk/modules/ble/src/bleio_seq_linux.c Func:BLEIO_Seq_Run Line:329 An error occurred while scheduling an instruction of type 0 for characteristic 00002A24-0000-1000-8000-00805F9B34FB
Error: Time:Tue Jan 10 05:45:11 2017 File:/home/pi/q/azure-iot-gateway-sdk/modules/ble/src/ble.c Func:on_connect_complete Line:553 BLEIO_Seq_Run failed

**Comment by avranju** [comment]
Could you please paste the output you're seeing when you run `list-attributes` from `bluetoothctl`?

**Comment by ahidaka** [comment]
Thanks. When I checked the list-attributes, there are no response messages. Your advice was helpful to understand the BLE gateway sample code.

After I reboot the system, "list-attributes" output was fixed normally. But I found new error massage for gateway execution. Anyway I will close this issue and open new issue for new error.

**Comment by jorgedfe** [comment]
ahidaka please can you help me? because, I have the same problem. How to fixed the error?

Error: Time:Thu Mar 30 22:03:42 2017 File:/home/sanitas01/azure-iot-gateway-sdk/modules/ble/src/ble_gatt_io_linux_read.c 
Func:BLEIO_gatt_read_char_by_uuid Line:151 g_tree_lookup() failed.

**Comment by ahidaka** [comment]
Note that I think that Bluez is not so stable at this time.
Please check this.
1. Reboot your board (Raspberry Pi 3?).
2. Stop and start bluetooth service as followings.
$sudo systemctl stop bluetooth
$sudo systemctl start bluetooth
$sudo rfkill unblock Bluetooth
3. Check the list-attibute result of bluetoothctl as followings.
$bluetoothctl
> power on
> scan on
> scan off
> connect <<your sensor-tag mac address>>
> list-attributes
if you cannot connect or receive list-attribute status, your BLE environment should be broken.

**Comment by ahidaka** [comment]
I will give you another advice. Check your gateway_sample.json file carefully.
"macAddress" and "device_mac_address" in json file should be the same.
The both are your Sensor-Tag address as followings.
      "args": [
        {
          "macAddress": "A0:??:??:??:??:??",

      "args": {
        "controller_index": 0,
        "device_mac_address": "A0:??:??:??:??:??",

**Comment by itctrial** [comment]
Same error here. 
Gateway is running.
Error: Time:Tue Jun 20 19:08:44 2017 File:/home/pi/iot-edge/modules/ble/src/ble_gatt_io_linux_read.c Func:BLEIO_gatt_read_char_by_uuid Line:151 g_tree_lookup() failed.
Error: Time:Tue Jun 20 19:08:44 2017 File:/home/pi/iot-edge/modules/ble/src/bleio_seq_linux_schedule_read.c Func:schedule_read Line:114 BLEIO_gatt_read_char_by_uuid failed with 149.
Error: Time:Tue Jun 20 19:08:44 2017 File:/home/pi/iot-edge/modules/ble/src/bleio_seq_linux.c Func:BLEIO_Seq_Run Line:342 An error occurred while scheduling an instruction of type 0 for characteristic 00002A24-0000-1000-8000-00805F9B34FB
Error: Time:Tue Jun 20 19:08:44 2017 File:/home/pi/iot-edge/modules/ble/src/ble.c Func:on_connect_complete Line:543 BLEIO_Seq_Run failed

**I tried to list-attributes under bluetoothctl but no attributes were shown. please help**

**Comment by guo-tt** [comment]
Is the board must be SensorTag board? I am using another kind of bluetooth board. But always have the error like this;

Gateway is running.

Error: .. ble_gatt_io_linux_read.c Func:BLEIO_gatt_read_char_by_uuid Line:151 g_tree_lookup() failed...

The bluetooth in raspberry pi can connect the board and can see the attributes in the board.
But the IoT gateway is failed to connect to the board..
![wechat image_20180327171000](https://user-images.githubusercontent.com/36875031/37959904-ce56e1b0-31e6-11e8-9074-4cefaa4d1140.jpg)
